convert2rhel/unit_tests/subscription_test.py

```diff
@@ -761,37 +766,118 @@ def test_args(self, registration_kwargs):
+        reg_cmd = subscription.RegistrationCommand(
+            org="justice_league",
+            activation_key="wonder twin powers",
+            rhsm_hostname=rhsm_hostname,
+            rhsm_port=rhsm_port,
+            rhsm_prefix=rhsm_prefix,
+        )
+        assert reg_cmd.connection_opts == expected
 
+    @pytest.mark.rhsm_returns(None)
+    def test_calling_registration_command_activation_key(self, monkeypatch, mocked_rhsm_register_with_activation_keys):
         reg_cmd = subscription.RegistrationCommand(activation_key="0xDEADBEEF", org="Local Organization")
-        assert reg_cmd() == ("", 0)
+        reg_cmd()
 
-        utils.run_subprocess.assert_called_once_with(
-            ["subscription-manager", "register", "--force", "--activationkey=0xDEADBEEF", "--org=Local Organization"],
-            print_cmd=False,
+        mocked_rhsm_register_with_activation_keys.assert_called_once_with(
+            "Local Organization",
+            ["0xDEADBEEF"],
+            dbus.Dictionary({"force": True}, "ss"),
+            dbus.Dictionary({}, "ss"),
+            "C",
+            dbus_interface="com.redhat.RHSM1.Register",
         )
-        assert utils.run_cmd_in_pty.call_count == 0
-
-    def test_calling_registration_command_password(self, monkeypatch):
-        monkeypatch.setattr(utils, "run_subprocess", mock.Mock(return_value=("", 0)))
-        monkeypatch.setattr(utils, "run_cmd_in_pty", mock.Mock(return_value=("", 0)))
 
+    @pytest.mark.rhsm_returns(None)
+    def test_calling_registration_command_password(self, monkeypatch, mocked_rhsm_register):
         reg_cmd = subscription.RegistrationCommand(username="me_myself_and_i", password="a password")
+
+        reg_cmd()
+
+        mocked_rhsm_register.assert_called_once_with(
+            "",
+            "me_myself_and_i",
+            "a password",
+            dbus.Dictionary({"force": True}, "ss"),
+            dbus.Dictionary({}, "ss"),
+            "C",
+            dbus_interface="com.redhat.RHSM1.Register",
+        )
+
+    @pytest.mark.rhsm_returns(None)
+    def test_calling_registration_command_with_connection_opts(
+        self, global_tool_opts, monkeypatch, mocked_rhsm_register
+    ):
+        reg_cmd = subscription.RegistrationCommand(
+            username="me_myself_and_i",
+            password="a password",
+            rhsm_hostname="rhsm.redhat.com",
+            rhsm_port="443",
+            rhsm_prefix="/",
+        )
+
+        run_subprocess_mocked = mock.Mock(return_value=("", 0))
+        monkeypatch.setattr(utils, "run_subprocess", run_subprocess_mocked)
+
         reg_cmd()
 
-        utils.run_cmd_in_pty.assert_called_once_with(
-            ["subscription-manager", "register", "--force", "--username=me_myself_and_i"],
-            expect_script=(("[Pp]assword: ", "a password\n"),),
-            print_cmd=False,
+        assert run_subprocess_mocked.call_count == 1
+        print(run_subprocess_mocked.call_args.args[0])
```

Comment:
I think you forgot a print here

Reply:
Thanks.  That code was failing on centos8 so I was trying to see what was going on.  I think I have it fixed and all the prints have been removed.